extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -0,0 +1,94 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import androidx.car.app.CarContext
+import androidx.lifecycle.Lifecycle
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * This is the main entry point for controlling the Mapbox car map surface.
+ *
+ * You can add the [MapboxCarMap] to your entire [androidx.car.app.Session] using the [Lifecycle].
+ * And then any [androidx.car.app.Screen] that is using a supported template will automatically
+ * show the Mapbox map. You can also, specify a [MapboxCarMap] for each individual
+ * [androidx.car.app.Screen] by constructing the [MapboxCarMap] with the screen Lifecycle.
+ *
+ * Supported templates include:
+ *  - [androidx.car.app.navigation.model.NavigationTemplate]
+ *  - [androidx.car.app.navigation.model.RoutePreviewNavigationTemplate]
+ *  - [androidx.car.app.navigation.model.PlaceListNavigationTemplate]
+ *
+ * Customize your [MapboxCarMap] with your own implementations of [MapboxCarMapObserver].
+ * Use the [registerObserver] and [unregisterObserver] functions to load and detach the observers.
+ *
+ * @param mapInitOptions to initialize the [MapboxCarMapSurface]
+ */
+@MapboxExperimental
+class MapboxCarMap(
```

Comment:
would the name be confused with MapboxMap?

i.e. 

`maboxCarMap.carMapSurface.mapSurface.getMapboxMap()`

Reply:
Do you have a suggestion? We have iterated a few times on this name and it seems good to me

It's the CarContext that has removed the potential confusion
``` kotlin
val carContext: CarContext = mapInitOptions.context as? CarContext
    ?: throw IllegalArgumentException("You must construct a MapboxCarMap with the CarContext")
```